test/gke: use correct cluster IPv4 CIDR

jibi opened the pull request:
Instead of hardcoding 10.0.0.0/8 as cluster IPv4 CIDR, which includes
also node IPs and may cause some tests to fail, use the correct one
obtained from the `gcloud container clusters describe` command.

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>

qmonnet commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -507,12 +507,17 @@ cluster.
 
+.. note:: The correct ``NATIVE_CIDR`` value can be obtained, as described in the
+          :ref:`k8s_install_gke` document, with the
+          ``gcloud container clusters describe $CLUSTER_NAME --zone $CLUSTER_ZONE --format 'value(clusterIpv4Cidr)')``
```
Would be great to have this line _above_ the ginkgo command. Otherwise, people may copy-paste the ginkgo command and fail to pass the native CIDR correctly, come back to the doc, search for the step they missed, etc.

What about inserting a step, or even simpler, just adding this command to the code block for the ginkgo command?

jibi replied:
Added the command to the code block :+1: 
I also kept the note (moved above) to have some more context on why those commands are needed

jibi committed:
test/gke: use correct cluster IPv4 CIDR

Instead of hardcoding 10.0.0.0/8 as cluster IPv4 CIDR, which includes
also node IPs and may cause some tests to fail, use the correct one
obtained from the `gcloud container clusters describe` command.

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>

qmonnet reviewed (approved):
Thanks!

nbusseneau reviewed (approved):
Cool! Do you think it would be worth to backport this, or nah?
